### DISCUSSION TITLE
Having a separate "components" and "pages" folder feels bad, man.  /:

### DISCUSSION THREAD

**Discussion by Johnrobmiller** [vercel/next.js#24044] [Ideas]
### Describe the feature you'd like to request

I'm aware that this is, technically speaking, a duplicate (https://github.com/vercel/next.js/issues/3728), but I think it's ok because the duplicates are a little bit old now.  

The organization system required by Next.js feels lousy for many people, myself included.  For reference, this is what I mean:
- pages
  - page1.tsx
  - page2.tsx
  - ...
  - page[x].tsx
- components
  - page1
    - [components]
  - page2
    - [components]
  - ...
  - page[x]
    - [components]

On the other hand, something like this would feel much nicer.  Let's pretend that any files in the "lib" folder do not get rendered as pages:
- pages
  - page1
    - index.tsx
    - lib
      - [components]
  - page2
    - index.tsx
    - lib
      - [components]
  - ...
  - page[x]
    - index.tsx
    - lib
      - [components]

Any updates on getting this implemented for 2021?  I hate to be "that guy" who complains on github forums, but this seems like something that would be very easy to program into Next.js  All the programmer would have to do is dynamically restructure the folder system of the 2nd example to be like the 1st when the code is being built.

Other than this one thing, I'm enjoying Next.js quite a bit, and cheers to the devs.

### Describe the solution you'd like

(explained in the above section)

### Describe alternatives you've considered

(explained in the above section)

**Comment by mnlfischer** [comment]
In a "component thinking way" what you should consider when using React, there is no need to separate components by pages. Do you have a specific use case why you do that? What about general components like buttons in your example?  For me it makes totally sense to use a page as the rendered outcome composition where all your specific components are used.

**Reply by Johnrobmiller** [reply]
To answer your question, I think that there might be a misunderstanding here; perhaps if I would have been more clear in the original post, it might have gotten less thumbs down.  In my project right now, I actually do have a "components" folder where I store all components that are reusable from page to page.  However, my project also has a /games route that routes to dozens of pages that each contain their own interactive css-based "mini-game".  For these unique pages, there are very few components that are reusable.  This is the reason why I suggested the feature above.

Technically speaking, the solution for my project is not quite the same as the one above.  I use: 

(in the root directory)
- components (for reuable components)
- states (for global states)
- page-files
  - games
    - game1
      - (non-reusable components, recoil states, js utility files, and css modules all go here)
    - game2
      -  (non-reusable components, recoil states, js utility files, and css modules all go here)
...and so on and so forth.

Also, regarding all the thumbs down, this is essentially the same solution as the one presented in the thread I linked to, which got no thumbs down.  Did I fuck up somehow in a different way that would explain all these thumbs down?  Did I post in the wrong section?  Are people not happy about how this is, technically speaking, a duplicate?  Did they need more context to get where this post was coming from (as mentioned above)?  Whenever I mess up in some way, it's usually good to know why I screwed up so I could not do that thing in the future, but I genuinely can't figure out where the thumbs downs are coming from.

**Reply by Johnrobmiller** [reply]
Oh, I just realized, the title is super misleading.  It's like I'm suggesting that there shouldn't be a folder containing re-usable components.  This was not my intention, and I'm guessing this is the reason this has so many thumbs down.
